Sources/Hummingbird/Application.swift

```diff
@@ -54,6 +53,119 @@ public final class HBApplicationContext: Sendable {
     }
 }
 
+public protocol HBApplicationProtocol: Service {
+    /// Context passed with HBRequest to responder
+    associatedtype Context: HBRequestContext
+    /// Responder that generates a response from a requests and context
+    associatedtype Responder: HBResponder<Context>
```

Comment:
I think this requires any `HBApplicationProtocol` implementation to explicitly specify the context type - however, `HBResponder` already defines it's own `Context` associated type, so that requirement could be dropped:

```suggestion
    /// Context passed with HBRequest to responder
    typealias Context = Responder.Context
    /// Responder that generates a response from a requests and context
    associatedtype Responder: HBResponder where Responder.Context: HBRequestContext
```

Of course the `typealias` isn't necessary per se, but might still be nice to have for ergonomics.

Reply:
There was a reason I did include the Context associated type, it does appear though I don't need it. It has been removed.